pkg/webui/console/containers/join-eui-prefixes-input/compute-prefix.js

```diff
@@ -0,0 +1,31 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const CHAR_BYTES = 4
+
+/**
+ * Computes the join EUI prefix given `joinEUI` and its `length`.
+ * @param {string} joinEUI - The join EUI.
+ * @param {number} length - The length of the prefix.
+ * @returns {string} - The join EUI prefix.
+ */
+function computePrefix(joinEUI, length = 0) {
+  const diff = length % CHAR_BYTES
+
+  // TODO: Support generating all prefixes that dont fulfil `length % CHAR_BYTES === 0`
+  // (https://github.com/TheThingsNetwork/lorawan-stack/issues/1263)
+  return joinEUI.slice(0, (length - diff) / CHAR_BYTES)
```

Comment:
We actually need to round this up instead of down. With this code, a prefix of `1111111111111111/15` gives `111`, allowing the user to choose something starting with `1114`, while only `1110` and `1111` are within `1111111111111111/15`.